RecoVertex/PrimaryVertexProducer/interface/HITrackFilterForPVFinding.h

```diff
@@ -25,14 +29,20 @@ class HITrackFilterForPVFinding : public TrackFilterForPVFinding {
     std::vector<reco::TransientTrack> seltks = TrackFilterForPVFinding::select(tracks);
     if (seltks.size() < NumTracksThreshold_) {
       return tracks;
-    } else {
-      return seltks;
+    } else if (seltks.size() > MaxNumTracksThreshold_) {
+      std::vector<reco::TransientTrack> seltksTight = TrackFilterForPVFinding::selectTight(tracks, minPtTight_);
+      if (seltksTight.size() >= NumTracksThreshold_)
+        return seltksTight;
     }
+
+    return seltks;
   }
 
   static void fillPSetDescription(edm::ParameterSetDescription& desc) {
     TrackFilterForPVFinding::fillPSetDescription(desc);
     desc.add<int>("numTracksThreshold", 0);  // HI only
+    desc.add<int>("maxNumTracksThreshold", 100000000);
```

Comment:
In the same spirit as above, what about `std::numeric_limits<int>::max()`?

Reply:
Also here, @abaty, could you please use `std::numeric_limits<int>::max()`?